addons/point_of_sale/point_of_sale.py

```diff
@@ -1464,6 +1464,37 @@ def insert_data(data_type, values):
             order.write({'state':'done', 'account_move': move_id})
 
         all_lines = []
+        if order.company_id.anglo_saxon_accounting:
```

Comment:
^ all_lines = [] should be after the new block.

+ extract the new code block into a new inner function e.g. add_anglosaxon_lines() to better separate it for readability (for what that's worth in such a long ugly piece of code)